webserver/cors/src/main/java/io/helidon/webserver/cors/CrossOriginConfig.java

```diff
@@ -0,0 +1,308 @@
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * @return the allowed origins
+     */
+    public String[] allowOrigins() {
+        return copyOf(allowOrigins);
+    }
+
+    /**
+     * @return the allowed headers
+     */
+    public String[] allowHeaders() {
+        return copyOf(allowHeaders);
+    }
+
+    /**
+     * @return headers OK to expose in responses
+     */
+    public String[] exposeHeaders() {
+        return copyOf(exposeHeaders);
+    }
+
+    /**
+     * @return allowed methods
+     */
+    public String[] allowMethods() {
+        return copyOf(allowMethods);
+    }
+
+    /**
+     * @return allowed credentials
+     */
+    public boolean allowCredentials() {
+        return allowCredentials;
+    }
+
+    /**
+     * @return maximum age
+     */
+    public long maxAge() {
+        return maxAge;
+    }
+
+    private static String[] copyOf(String[] strings) {
+        return strings != null ? Arrays.copyOf(strings, strings.length) : new String[0];
+    }
+
+    /**
+     * Builder for {@link CrossOriginConfig}.
+     */
+    public static class Builder implements Setter<Builder>, io.helidon.common.Builder<CrossOriginConfig>,
+            Function<Config, Builder> {
+
+        static final String[] ALLOW_ALL = {"*"};
+
+        private String pathPrefix = PATHLESS_KEY; // not typically used except when inside a MappedCrossOriginConfig
+        private boolean enabled = true;
+        private String[] origins = ALLOW_ALL;
+        private String[] allowHeaders = ALLOW_ALL;
+        private String[] exposeHeaders;
+        private String[] allowMethods = ALLOW_ALL;
+        private boolean allowCredentials;
+        private long maxAge = DEFAULT_AGE;
+
+        private Builder() {
+        }
+
+        /**
+         * Creates a new builder based on the values in an existing {@code CrossOriginConfig} object.
+         *
+         * @param original the existing cross-origin config object
+         * @return new Builder initialized from the existing object's settings
+         */
+        public static Builder from(CrossOriginConfig original) {
+            return new Builder()
+                    .pathPrefix(original.pathPrefix)
+                    .enabled(original.enabled)
+                    .allowCredentials(original.allowCredentials)
+                    .allowHeaders(original.allowHeaders)
+                    .allowMethods(original.allowMethods)
+                    .allowOrigins(original.allowOrigins)
+                    .exposeHeaders(original.exposeHeaders)
+                    .maxAge(original.maxAge);
+        }
+
+        /**
+         * Creates a new {@code Builder}instance from the specified configuration.
+         *
+         * @param config node containing cross-origin information
+         * @return new {@code Builder} initialized from the config
+         */
+        public static Builder from(Config config) {
```

Comment:
Do not add static factory methods to builders.
Please move line 238 to the static `builder` method on `CrossOriginConfig`

Reply:
See below.